tests/dependency_analyzer/configurations/micronaut_core_config.yaml

```diff
@@ -13,13 +13,13 @@ dependencies:
 - id: slf4j
     # For version 1.7.36
     # https://github.com/qos-ch/slf4j/commit/e9ee55cca93c2bf26f14482a9bdf961c750d2a56
-  branch: v_1.7.36
+  branch: 1.7-maintenance
```

Comment:
I wonder why you need to update the branches in these `yaml` input files for the integration tests 🤔.

Reply:
The reason I made this changes is because `v_1.7.36` is a tag, but not a branch. In the old implementation of the check out logic, the `branch` value here is used in `git checkout v_1.7.36`. This check out is OKAY because we can checkout a tag in git normally -> we missed this error.
However, my intention with this part in the first place is that: Macaron only checkouts the commit if it belong to the branch name provided in `branch`  in the yaml config file.
Therefore, with the new implementation where the commit is explicitly checked if it is contained within `branch`, the value `v_1.7.36` is flagged as error. It has been updated here with the actually branch name `1.7-maintenance` instead.